lib/migrations/conditions.go

```diff
@@ -0,0 +1,53 @@
+package migrations
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const deprecatedPipelineCompletedCondition = "PipelineCompleted"
+
+func RemoveDeprecatedConditions(ctx context.Context, k8sClient client.Client, obj *unstructured.Unstructured, logger logr.Logger) (*ctrl.Result, error) {
+	conditions, found, err := unstructured.NestedSlice(obj.Object, "status", "conditions")
+	if err != nil || !found {
+		logger.Info("No deprecated conditions found on resource", "resource", obj.GetName())
+		return nil, nil
+	}
+
+	newConditions := filterOutCondition(conditions, deprecatedPipelineCompletedCondition)
+	if len(newConditions) == len(conditions) {
+		return nil, nil
+	}
+
+	logger.Info("Removing deprecated condition", "conditionType", deprecatedPipelineCompletedCondition)
+	if err := unstructured.SetNestedSlice(obj.Object, newConditions, "status", "conditions"); err != nil {
+		logger.Error(err, "failed to remove deprecated conditions on resource")
+		return nil, err
+	}
+
+	// Update the status using the client
+	if err := k8sClient.Status().Update(ctx, obj); err != nil {
```

Comment:
```suggestion
	if err = unstructured.SetNestedSlice(obj.Object, newConditions, "status", "conditions"); err != nil {
		logger.Error(err, "failed to remove deprecated conditions on object", "name", obj.GetName(), "kind", obj.GetKind())
		return nil, err
	}

	if err = k8sClient.Status().Update(ctx, obj); err != nil {
```

Mostly because this can be a promise or a user resource, and resource can be interpreted as a kratix resource.